classes/submission/Submission.inc.php

```diff
@@ -248,6 +248,42 @@ function setHideAuthor($hideAuthor) {
 	}
 
 	/**
+	 * Set submission revision.
+	 * @param $submissionRevision int
+	 */
+	function setSubmissionRevision($submissionRevision) {
+		$this->setData('submissionRevision', $submissionRevision);
+	}
+
+	/**
+	 * Return submission revision.
+	 * @return int
+	 */
+	function getSubmissionRevision() {
+		return $this->getData('submissionRevision');
+	}
+
+	/**
+	 * Get the current revision id for a submission
+	 * @param $contextId int
+	 * @return int
+	 */
+	function getCurrentVersionId() {
+		$submissionId = $this->getId();
+		$contextId = $this->getContextId();
+
+		// handle publisher ids
+		if (!ctype_digit("$submissionId")) {
```

Comment:
I cannot believe that getId() is special for submissions -- we are using it everywhere :-)
You could print $submissionId in a file before this line, for example, and then do something that would call this function, then take a look at the file. For example:
$file = 'debug.txt';
$current = file_get_contents($file);
$current .= print_r($submissionId, true);
file_put_contents($file, $current);
Would print the submission id in the file debug.txt...